src/js/camresolutionstest.js

```diff
@@ -36,148 +20,109 @@ var resolutions = [[160, 120, false],
  * like average encode time and FPS. A test case will be created per mandatory
  * resolution found in the "resolutions" array.
  */
-for (var index = 0; index < resolutions.length; index++) {
-  if (resolutions[index][2]) {
-    var testTitle = testCaseName.CHECKRESOLUTION + resolutions[index][0] + 'x' +
-                    resolutions[index][1];
-    addTest(testSuiteName.CAMERA, testTitle,
-        cameraTest_.bind(null, [resolutions[index]]));
-  }
-}
 
-/*
- * "Supported resolutions" test tries calling getUserMedia() with each
- * resolution from the list below. Each gUM() call triggers a success or a fail
- * callback; we report ok/nok and schedule another gUM() with the next
- * resolution until the list is exhausted. Some resolutions are mandatory and
- * make the test fail if not supported.
- */
-addTest(testSuiteName.CAMERA, testCaseName.CHECKSUPPORTEDRESOLUTION,
-    cameraTest_.bind(null, resolutions));
+addTest(testSuiteName.CAMERA, testCaseName.CHECKRESOLUTION240, function(test) {
+  var camResolutionsTest = new CamResolutionsTest(test);
+  camResolutionsTest.run([320, 240]);
+});
 
-function cameraTest_(resolutions) {
-  var test = new CamResolutionsTest(resolutions);
-  test.run();
-}
+addTest(testSuiteName.CAMERA, testCaseName.CHECKRESOLUTION480, function(test) {
+  var camResolutionsTest = new CamResolutionsTest(test);
+  camResolutionsTest.run([640, 480]);
+});
+
+addTest(testSuiteName.CAMERA, testCaseName.CHECKRESOLUTION720, function(test) {
+  var camResolutionsTest = new CamResolutionsTest(test);
+  camResolutionsTest.run([1280, 720]);
+});
 
-function CamResolutionsTest(resolutionArray) {
-  this.resolutions = resolutionArray;
+function CamResolutionsTest(test) {
+  this.test = test;
   this.mandatoryUnsupportedResolutions = 0;
-  this.numResolutions = this.resolutions.length;
   this.counter = 0;
   this.supportedResolutions = 0;
   this.unsupportedResolutions = 0;
   this.currentResolutionForCheckEncodeTime = null;
-
   this.isMuted = false;
-  this.stream = null;
-}
-
-function resolutionMatchesIndependentOfRotationOrCrop_(aWidth, aHeight,
-                                                       bWidth, bHeight) {
-  var minRes = Math.min(bWidth, bHeight);
-  return (aWidth === bWidth && aHeight === bHeight) ||
-         (aWidth === bHeight && aHeight === bWidth) ||
-         (aWidth === minRes && bHeight === minRes);
 }
 
 CamResolutionsTest.prototype = {
-  run: function() { this.triggerGetUserMedia_(this.resolutions[0]); },
-
-  triggerGetUserMedia_: function(resolution) {
+  run: function(resolution) {
```

Comment:
I will structure it the same way as the other tests ;) once we have decided if we should keep the old resolution checks.